brion/plugin/morphologyHDF5.cpp

```diff
@@ -43,320 +43,351 @@ lunchbox::PluginRegisterer<MorphologyHDF5> registerer;
+                                       "': bad number of dimensions in"
+                                       " 'structure' dataspace"));
+        }
     }
 
-    bool readV11Metadata()
+    bool _readV11Metadata()
     {
         try
         {
-            const H5::Group& metadata = file.openGroup(_g_metadata);
-            const H5::Attribute& attr = metadata.openAttribute(_a_version);
+            detail::SilenceHDF5 silence;
+            const auto metadata = _file->getGroup(_g_metadata);
+            const auto attr = metadata.getAttribute(_a_version);
 
             uint32_t version[2];
-            attr.read(H5::PredType::STD_U32LE, &version[0]);
+            attr.read(version);
             if (version[0] != 1 || version[1] != 1)
                 return false;
 
-            initData.version = MORPHOLOGY_VERSION_H5_1_1;
+            _initData.version = MORPHOLOGY_VERSION_H5_1_1;
 
-            const H5::Attribute& familyAttr = metadata.openAttribute(_a_family);
-            H5::EnumType familyEnum = metadata.openEnumType(_e_family);
-
-            familyAttr.read(familyEnum, &initData.family);
-
-            resolveV1();
-            return true;
+            const auto familyAttr = metadata.getAttribute(_a_family);
+            uint32_t family;
+            familyAttr.read(family);
+            _initData.family = (CellFamily)family;
         }
-        catch (const H5::Exception&)
+        catch (const HighFive::GroupException&)
         {
             return false;
         }
+        catch (const HighFive::Exception& e)
+        {
+            // All other exceptions are not expected because if the metadata
+            // group exits it must contain at least the version, and for
+            // version 1.1 it must contain the family.
+            LBTHROW(std::runtime_error(
+                std::string("Error reading morphology metadata: ") + e.what()));
+        }
+
+        _resolveV1();
+        return true;
     }
 
-    bool readV2Metadata()
+    bool _readV2Metadata()
     {
         try
         {
-            file.openGroup(_g_root);
-            initData.version = MORPHOLOGY_VERSION_H5_2;
-            return true;
+            detail::SilenceHDF5 silence;
+            const auto root = _file->getGroup(_g_root);
+            const auto attr = root.getAttribute(_a_version);
+            attr.read(_initData.version);
+            if (_initData.version == MORPHOLOGY_VERSION_H5_2)
+                return true;
         }
-        catch (const H5::Exception&)
+        catch (const HighFive::Exception&)
         {
-            return false;
         }
-    }
 
-    void resolveV1()
-    {
-        pointDS = file.openDataSet("/" + _d_points);
-        const H5::DataSpace pointsSpace = pointDS.getSpace();
-        if (pointsSpace.getSimpleExtentNdims() != 2 ||
-            pointsSpace.getSimpleExtentDims(pointsDims) < 0 ||
-            pointsDims[1] != _pointColumns)
+        try
```

Comment:
I think you brought back the old code which is after some conclusion the same as  what Stefan did, so why again adding 10+ more lines here? Unless I miss something again. Hard to review with the all the reshuffling here.

Reply:
Not sure what you mean by bringing the old code back. Consider that I had done these modifications days before Stefan changes, so it's not a surprise that it may remind you to the original code.